BASE/src/ServerTelemetryChannel/ServerTelemetryChannel.cs

```diff
@@ -310,6 +311,29 @@ public void Flush()
             this.TelemetryBuffer.FlushAsync().ConfigureAwait(false).GetAwaiter().GetResult(); // Don't use Task.Wait() because it wraps the original exception in an AggregateException.
         }
 
+        /// <summary>
+        /// Asynchronously flushes the telemetry buffer. 
+        /// </summary>
+        /// <returns>
+        /// True indicates telemetry data ownership is transferred out of process, that are emitted before the flush invocation.
+        /// False indicates transfer of telemetry data has failed, the process still owns all or part of the telemetry.
+        /// </returns>
+        public Task<bool> FlushAsync(CancellationToken cancellationToken)
+        {
+            if (!this.isInitialized)
+            {
+                TelemetryChannelEventSource.Log.StorageNotInitializedError();
+            }
+
+            TelemetryChannelEventSource.Log.TelemetryChannelFlushAsync();
+
+#if NET45
+            return cancellationToken.IsCancellationRequested ? Task.Factory.StartNew(() => false, cancellationToken) : this.TelemetryBuffer.ManualFlushAsync(cancellationToken);
+#else
+            return cancellationToken.IsCancellationRequested ? Task.FromCanceled<bool>(cancellationToken) : this.TelemetryBuffer.ManualFlushAsync(cancellationToken);
+#endif
```

Comment:
This pattern appears 3 times in the PR. It's not exactly the same each time, but it is all for the lack of `Task.FromCanceled` in NET45.

I know I'm repeating myself, but I suggest creating a helper method (or maybe 2 if we need the non-generic version too) that matches the signature of `Task.FromCanceled` and can be called from here and the other 2 places without using conditional compilation (`#if / #else / #endif`).

For example:
```csharp
internal static class TaskEx
{
    public static Task FromCanceled(CancellationToken cancellationToken)
    {
#if NET45
        // TODO: Equivalent for .NET 4.5
#else
        return Task.FromCanceled(cancellationToken);
#endif  
    }

    public static Task<TResult> FromCanceled<TResult>(CancellationToken cancellationToken)
    {
#if NET45
        // TODO: Equivalent for .NET 4.5
#else
        return Task.FromCanceled<TResult>(cancellationToken);
#endif  
    }
}
```